### ISSUE TITLE
Spurious routing fails (only on page reload)

### ISSUE THREAD

**Issue by c-bik** [krasimir/navigo#239]
Hi,

Thank you very much for the awesome job with navigo!

I am using it a project for quite a while now in prodection. Everything is running smoothly so far.

Recently I am trying to change `#!` to full histry API support and got stuck with a routing problem.

**Version : `7.1.2`**

I have the following routes:
1. `/link1/:year`
1. `/link2/:year`
1. `/contact`
1. `/about`
1. `*`

### `index.js`
```js
const
    Navigo = require('navigo'),
    router = new Navigo();

router.hooks({
    before: (done, params) => {
        console.log('navigo', window.location, 'before_hook', params);
        done();
    },
    after: (params) => {
        console.log('navigo', window.location, 'after_hook', params);
    }
});

router.on({
        '*': () => { console.log("home"); }
        },
        '/link1/:year/': (params) => {
            console.log("link1/" + params.year);
        },
        '/link2/:year': (params) => {
            console.log("link2/" + params.year);
        },
        '/contact': () => {
            console.log("contact");
        },
        '/about': () => {
            console.log("about");
        }
    }).resolve();
```

It is all working prefectly when is page is loaded.
Also when I try to refresh / reload browser with this nginx config static contents (html/js) are loading correctly (ignoring the virtual paths `link1`, `contact` etc):
###  `nginx.conf`
```conf
location ~ ^/(about|contact|link1/[0-9]+|link2/[0-9]+)(.*)$ {
    alias /mnt/c/projects/git/KonnexionsGmbH/nnedpro/eportal/public/$2;
}
```

`navigo` is correctly routing for:
1. `/contact`
1. `/about`
1. `*`
**working in both cases of client side routing or page reload**

## Problem
But the following are also routed to `*`
1. `/link1/:year`
1. `/link2/:year`

To understand what is going I added router hooks `router.hooks`. And this what I am seeing in console logs (Chrome) when the route failure on page reload happens (re-formatted and summarized):

```js
// index.js:25
navigo Location {
    href: "https://127.0.0.1/link1/2019/",
    origin: "https://127.0.0.1",
    protocol: "https:",
    …
}
before_hook null

// index.js:100
home

// index.js:29
navigo Location {
    href: "https://127.0.0.1/link1/2019/",
    origin: "https://127.0.0.1",
    protocol: "https:",
    …
}
after_hook null
```
### Only `navigo` routing (no page reload)
```js
// index.js:25
navigo Location {
    href: "https://127.0.0.1/link1/2019",
    origin: "https://127.0.0.1",
    protocol: "https:",
    …
}
before_hook {year: "2019"}

// index.js:108
summer_schools/2019

// index.js:29
navigo Location {
    href: "https://127.0.0.1/link1/2019",
    origin: "https://127.0.0.1",
    protocol: "https:",
    …
}
after_hook {year: "2019"}
```

**It seems like (I may be very wrong!) that the trailing `/` (inserted by `nginx`) is causing this. If that is the case, I don't know how to get rid of that from `nginx` or how to match to it in `navigo`. Also the trailing `/` is only seems to be the problem when the route has parameter matching (`:year`)**

This issue is probably not a duplicate of https://github.com/krasimir/navigo/issues/214 as I am not using `webpack-dev-server` etc!

It feels like I am missing something silly but I don't seem to figure out what it is.

Can you please help?

**Comment by c-bik** [comment]
@krasimir Can you please help me out?

**Comment by krasimir** [comment]
Hello,

this seems weird to me:
```
'/link1/:year/': (params) => {
 ...
},
'/link2/:year': (params) => {
 ...
}
```
What is the use case and is it possible to use `/link1/:year/<something here>`?

**Comment by c-bik** [comment]
> Hello,
> 
> this seems weird to me:
> 
> ```
> '/link1/:year/': (params) => {
>  ...
> },
> '/link2/:year': (params) => {
>  ...
> }
> ```
> 
> What is the use case and is it possible to use `/link1/:year/<something here>`?

Hi @krasimir,

First off thanks for replying.

Let me try to examplain my use case first. Perhaps you can propose a better design than what I have.

There are two independent sites (wordpress) which publishes one page every year with static info relevant for that year and my site archieves those pages. Hence I have `link1/:year` and `/link2/:year`. From the `:year` part I load the HTML from the archieve `requite('link1/2005.html')` (for example) and the same for the other link.

This design was the simpliest I could think of. These URL will not (for now) have anything after `:year/` so there nothing for `<something here>`.

How should I go about it?

TIA

**Comment by c-bik** [comment]
@krasimir (sorry for poking you again) do you have any ideas?

**Comment by antiv** [comment]
HI, I had same problem.
I have simple routes and one with params like `/content/:id` On refresh, or when I open that url directly I got error. SO, I changed 
```navigoRouter = new Navigo(null, useHash)``` 
to
 ```navigoRouter = new Navigo(getRootLocation(routes), useHash)```, and add method 
```
const getRootLocation = (routes): string => {
    let winPath = window.location.pathname
    Object.entries(routes).forEach(([path, route]) => {
      const pathParts = path.split(':')
      winPath = winPath.split(pathParts[0])[0]
    })
    const root = window.location.protocol + '//' + window.location.host + winPath
    console.log('Site root', root)
    return root
  }
```
so, now it works in server root, and from subfolder. 
And, you need to define base tag in index.htm also

**Comment by krasimir** [comment]
@c-bik @antiv sorry for not responding here. I abandoned the project for a couple of reasons but now I'm back with a big re-write. There is a version 8.0.0 which I believe will solve your problems. I know that it's being more than a year since you write here but if you still use Navigo and want to try a fix install the new one.  

Version 8.0.0 is not officially released but can be installed via `npm install navigo@beta` or `yarn add navigo@beta`. Migration guide is available here https://github.com/krasimir/navigo/blob/big-rewrite/CHANGELOG.md#migration-guide and the new documentation here https://github.com/krasimir/navigo/blob/big-rewrite/DOCUMENTATION.md.

I wanna start fresh so I'm not going to provide support for the 7.2.1 but only for the new 8.0.0+. In favor of this idea I'll close your issue but @c-bik please comment if you still experiencing problems and I'll reopen it.
